File: dart_native/example/lib/android/ConstructorTest.dart
import 'dart:ffi';

import 'package:dart_native/dart_native.dart';

class ConstructorTest extends JObject {

  ConstructorTest.alongXAxis([Pointer ptr]) : this("", 0, 0, false);

  ConstructorTest(String s, int i, int b, bool bo, [Pointer ptr]): super("com/dartnative/dart_native_example/ConstructorTest", ptr, [s, i, byte(b), bo]);

}

File: dart_native/example/android/app/src/main/java/com/dartnative/dart_native_example/ConstructorTest.java
package com.dartnative.dart_native_example;

import android.util.Log;

public class ConstructorTest {
    private static final String TAG = "ConstructorTest";

    public ConstructorTest() {
        this("", 0);
    }

    public ConstructorTest(String s, int i) {
        this(s, i, (byte)0, false);
    }

    public ConstructorTest(String s, int i, byte b, boolean bool) {
        Log.d(TAG, "constructor " + s + " " + i + " " + b + " " + bool);
    }
}
